File: FrontEnd/src/app/features/billing/invoicesViewer/classes/view-models/pdf/invoice-pdf-vm.ts
import { InvoicePdfPartyTypeVM } from './invoice-pdf-partyType-vm'
import { InvoicePdfHeaderVM } from './invoice-pdf-header-vm'
import { InvoicePdfSummaryVM } from './invoice-pdf-summary-vm'
import { InvoicePdfAadeVM } from './invoice-pdf-aade-vm'
import { InvoicePdfPortVM } from './invoice-pdf-port-vm'

export interface InvoicePdfVM {

    header: InvoicePdfHeaderVM
    issuer: InvoicePdfPartyTypeVM
    counterPart: InvoicePdfPartyTypeVM
    summary: InvoicePdfSummaryVM
    aade: InvoicePdfAadeVM
    ports: InvoicePdfPortVM[]

}

File: API/Features/Billing/Invoices/ViewModels/Pdf/InvoicePdfVM.cs
using System;
using System.Collections.Generic;

namespace API.Features.Billing.Invoices {

    public class InvoicePdfVM {

        public Guid InvoiceId { get; set; }
        public InvoicePdfHeaderVM Header { get; set; }
        public string Remarks { get; set; }
        public InvoicePdfShipVM Ship { get; set; }
        public InvoicePdfAadeVM Aade { get; set; }
        public List<InvoicePdfPortVM> Ports { get; set; }
        public InvoicePdfPartyVM Customer { get; set; }
        public string Destination { get; set; }
        public InvoicePdfDocumentTypeVM DocumentType { get; set; }
        public string PaymentMethod { get; set; }
        public InvoicePdfPartyVM Issuer { get; set; }
        public InvoicePdfSummaryVM Summary { get; set; }
        public decimal PreviousBalance { get; set; }
        public decimal NewBalance { get; set; }
        public string[] BankAccounts { get; set; }

    }

}